### DISCUSSION TITLE
Transparency to the Search User

### DISCUSSION THREAD

**Discussion by RJGhugo** [searxng/searxng#1244] [Ideas]
Hello,

I am very happy with my **searxng** installation!

At the bottom of the search-page there are serveral links.

On of them is showing some information about recent search requests (Suchmaschinenstatistiken).

I played around with settings #1241.

It would be nice to deliver "Hostname replace"-Plugin settings to the user for transparency.

e.g. This search has filtered out: "amazon, facebook, tripadvisor"...

**Comment by tiekoetter** [comment]
Do you want the user to turn hostname replace on or off? If yes then there is already an option under preferences -> general-> hostname replace

**Reply by RJGhugo** [reply]
No. I just want to inform the user somehow, that I am filtering the results, so that they are transparently informed about the "manipulation". Instead of some "endless lists of regular expression" it could also be a chart saying something like: "amazon: 10 x filtered out", "facebook: 20 x filtered out", "XYZABC: 50 x filtered out"...

**Reply by return42** [reply]
IDK if it is a good idea to see (count) URLs blocked by hostname_replace:

.. if you show this information to the end user what will this information say to the user and what will be the next question of the user .. and whats about redirected URLs in web and media results (video- and image-URLs) .. what about filters like the adult filter (aka safe-search) or the region filter .. 

By example: my instance is optimized for me and my *german speaking* friends (e.g. I have a category "TV" with the public service broadcasting from Germany) to get a maximum on privacy .. I filter out as much junk as possible, I filter out obscene and hateful content whenever I can .. and I use every method and technology I have for.

What I mean: there are many reasons to block results and I do not yet see the benefit for the user to count just particular ones while hiding other content without a comment.

TBH: no search engine I know informs the user about the results it does not show to the user  (depended on user's region, user's language,  user's IP, user's search history, ....).

If maintainers want to inform the users about the setup of the instance (and why/what content is not in the result list) they should describe their setups and intentions in an additional info page.

Info-Pages can be easily added in:  https://github.com/searxng/searxng/tree/master/searx/infopage

**Comment by return42** [comment]
> It would be nice to deliver "Hostname replace"-Plugin settings to the user for transparency.

This is what I thought also, but TBH I don't know if end-users will understand regular expressions .. and if it is helpful for an end-user to see endless lists of regular expression like I have on my instance: https://github.com/searxng/searxng/discussions/970

In the future I plan to make hostname_replace more generic .. e.g. loading regular expressions from trusted (configurable) sources similar to what we know from DNS blockers or from uBlock origin.

For me it becomes more and more questionable if it makes sense to show the setup of hostname_replace in the UI ..

**Reply by RJGhugo** [reply]
Similar to Pi-Hole?

**Comment by RJGhugo** [comment]
Thanks very much for the deep explication!
My conclusion:
For the usability I shall use the clean fast default page.
Meta search engines, by definition, already getting filtered results. So I do not have to explain what I do with the results anyway...

**Comment by dalf** [comment]
For reference : https://github.com/searxng/searxng/issues/765
